src/components/Marketplace/ConfirmOrderModal.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+
+import { dispatch } from 'store/rematch';
+import { selectModalOptions, selectOrderDetails, selectTokenDetails } from 'store/selectors';
+import { BitgoAction, sendToBitgo } from 'util/bitgoHelper';
+import { V } from 'util/theming';
+import { Colors, FEE, ModalName, TICKER } from 'vars/defines';
+
+import { CenteredButtonWrapper } from 'components/_General/_UIElements/common';
+import { Button } from 'components/_General/buttons';
+import { Column, Columns } from 'components/_General/Grid';
+import WarningCritical from 'components/_General/WarningCritical';
+import AssetWidget from './common/AssetWidget';
+import KeyValueDisplay from './common/KeyValueDisplay';
+
+const WarningWrapper = styled.div`
+  margin-top: 15px;
+  padding: 18px;
+  border-radius: ${V.size.borderRadius};
+  background-color: ${V.color.backHard};
+  margin-bottom: 20px;
+
+  & > [WarningCritical] {
+    margin-bottom: none;
+  }
+`;
+
+interface ConfirmOrderModalProps {}
+
+const ConfirmOrderModal: React.FC<ConfirmOrderModalProps> = () => {
+  const formValues = useSelector(selectModalOptions) as Record<string, any>;
+  const orderDetails = useSelector(selectOrderDetails);
+  const tokenDetails = useSelector(selectTokenDetails);
+
+  const currentOrderDetails = orderDetails?.[formValues?.orderId];
+  const currentOrderType = currentOrderDetails?.type;
+  const currentTokenDetails = currentOrderDetails?.token || tokenDetails?.[formValues?.assetId];
+  const isFilling = formValues?.type === 'fill';
+
+  const handleOrderBroadcast = () => {
+    if (formValues?.type === 'fill') {
+      sendToBitgo(
+        currentOrderType === 'bid' ? BitgoAction.ASSET_V2_FILL_BID : BitgoAction.ASSET_V2_FILL_ASK,
```

Comment:
Just a note:

All the `fill` `bid` `ask` should prolly live in a separate file as consts as they repeat.